# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteVideoDecoder.cpp
# Commit: c9f9bba02acf
# Full Hash: c9f9bba02acf40a824a92fd9430cc5e10edcc485
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-06-28 16:12:48
# Regressor Bug: 1561178
# File Overlap Count: 1
# Description:
#   Bug 1561178 - Make passing a TextureFactoryIdentifier optional, and pass one whenever we can. r=jya
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35965
# ==============================================================================

diff -r c633a515314d -r c9f9bba02acf dom/media/ipc/RemoteVideoDecoder.cpp
--- a/dom/media/ipc/RemoteVideoDecoder.cpp	Fri Jun 28 06:31:59 2019 +0000
+++ b/dom/media/ipc/RemoteVideoDecoder.cpp	Fri Jun 28 06:32:16 2019 +0000
@@ -151,7 +151,8 @@
 
 MediaResult RemoteVideoDecoderChild::InitIPDL(
     const VideoInfo& aVideoInfo, float aFramerate,
-    const CreateDecoderParams::OptionSet& aOptions) {
+    const CreateDecoderParams::OptionSet& aOptions,
+    const layers::TextureFactoryIdentifier* aIdentifier) {
   RefPtr<RemoteDecoderManagerChild> manager =
       RemoteDecoderManagerChild::GetRDDProcessSingleton();
 
@@ -174,9 +175,8 @@
   nsCString blacklistedD3D11Driver;
   nsCString blacklistedD3D9Driver;
   VideoDecoderInfoIPDL decoderInfo(aVideoInfo, aFramerate);
-  TextureFactoryIdentifier defaultIdent;
   if (manager->SendPRemoteDecoderConstructor(
-          this, decoderInfo, aOptions, defaultIdent, &success,
+          this, decoderInfo, aOptions, ToMaybe(aIdentifier), &success,
           &blacklistedD3D11Driver, &blacklistedD3D9Driver, &errorDescription)) {
     mCanSend = true;
   }
@@ -242,7 +242,7 @@
   nsCString errorDescription;
   VideoDecoderInfoIPDL decoderInfo(aVideoInfo, aFramerate);
   if (manager->SendPRemoteDecoderConstructor(
-          this, decoderInfo, aOptions, aIdentifier, &success,
+          this, decoderInfo, aOptions, Some(aIdentifier), &success,
           &mBlacklistedD3D11Driver, &mBlacklistedD3D9Driver,
           &errorDescription)) {
     mCanSend = true;
@@ -255,14 +255,14 @@
 RemoteVideoDecoderParent::RemoteVideoDecoderParent(
     RemoteDecoderManagerParent* aParent, const VideoInfo& aVideoInfo,
     float aFramerate, const CreateDecoderParams::OptionSet& aOptions,
-    const layers::TextureFactoryIdentifier& aIdentifier,
+    const Maybe<layers::TextureFactoryIdentifier>& aIdentifier,
     TaskQueue* aManagerTaskQueue, TaskQueue* aDecodeTaskQueue, bool* aSuccess,
     nsCString* aErrorDescription)
     : RemoteDecoderParent(aParent, aManagerTaskQueue, aDecodeTaskQueue),
       mVideoInfo(aVideoInfo) {
-  if (XRE_IsGPUProcess()) {
+  if (XRE_IsGPUProcess() && aIdentifier) {
     mKnowsCompositor = new KnowsCompositorVideo();
-    mKnowsCompositor->IdentifyTextureHost(aIdentifier);
+    mKnowsCompositor->IdentifyTextureHost(aIdentifier.value());
   }
 
   CreateDecoderParams params(mVideoInfo);